app/src/main/java/org/wikipedia/suggestededits/provider/EditingSuggestionsProvider.kt

```diff
@@ -366,41 +372,55 @@ object EditingSuggestionsProvider {
                     revertCandidateLastRevId = 0L
                 }
                 revertCandidateLang = lang
-                if (!revertCandidateCache.empty()) {
-                    L.d(revertCandidateCache.toString())
-                    cachedItem = revertCandidateCache.pop()
+                if (!revertCandidateCache.isEmpty()) {
+                    cachedItem = revertCandidateCache.removeFirst()
                 }
 
                 if (cachedItem == null) {
                     while (this.coroutineContext.isActive) {
                         try {
-                            val recentChanges = SuggestedEditsRecentEditsViewModel.getRecentEditsCall(WikiSite.forLanguageCode(lang), 10, null, mutableListOf())
-                                .first.sortedByDescending { it.curRev }
-                            var maxRevId = 0L
+                            // If we have been reset, then fetch a few *older* changes, so that the user
+                            // has a few changes to flip through. Otherwise, start fetching *newer* changes,
+                            // starting from the last recorded timestamp.
+                            val triple = if (revertCandidateLastRevId == 0L)
+                                SuggestedEditsRecentEditsViewModel.getRecentEditsCall(WikiSite.forLanguageCode(lang),
```

Comment:
Would it be better if we move `WikiSite.forLanguageCode(lang)` out of the `while`?

Reply:
sure!